js/main.js

```diff
@@ -0,0 +1,119 @@
+function Finder(status) {
+  this.status = 'home-page';
+}
+
+var holiday = new Finder();
+
+var homePage = document.querySelector('.home-page');
+var searchPage = document.querySelector('.search-page');
+var resultPage = document.querySelector('.result-page');
+
+var findButton = document.querySelector('.home-search');
+var submit = document.querySelector('.submit');
+var dataOutline = document.querySelector('.data-outline');
+var back = document.querySelector('.back');
+
+var searchBox = document.querySelector('#search-box');
+
+var backHome = ['search-page'];
+
+var main = document.querySelector('main');
+
+Finder.prototype.changePage = function (event) {
+  if (event.target === back && holiday.status === 'home-page') {
+    return;
+  }
+
+  if (event.target === searchBox) {
+    return;
+  }
+
+  homePage.className = 'home-page hidden';
+  searchPage.className = 'search-page hidden';
+  resultPage.className = 'result-page hidden';
+  if (event.target === findButton) {
+    searchPage.className = 'search-page';
+    holiday.status = 'search-page';
+  } else if (event.target === submit) {
+    dataOutline.innerHTML = '';
+    holiday.renderResults();
+    resultPage.className = 'result-page';
+    holiday.status = 'result-page';
+  } else if (event.target === back) {
+    for (var i = 0; backHome.length > i; i++) {
+      if (backHome[i] === holiday.status) {
+        homePage.className = 'home-page';
+        holiday.status = 'home-page';
+      }
+    }
+    if (holiday.status === 'result-page') {
+      searchPage.className = 'search-page';
+      holiday.status = 'search-page';
+    }
+  }
+};
```

Comment:
I'd suggest adding or removing the `'hidden'` class with `classList.prototype.add()` and `classList.prototype.remove()` (https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) over changing the whole `className` string.